core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -1,27 +1,225 @@
 		this.fullName = fullName;
 		this.simpleName = simpleName;
+		this.isAbstract = isAbstract;
+		this.aliasSources = new ArrayList<>();
+	}
+
+	public FrankElement(final String fullName, final String simpleName) {
+		this(fullName, simpleName, false);
+	}
+	
+	public void setParent(FrankElement parent) {
+		this.parent = parent;
+		this.statistics = new FrankElementStatistics(this);
+	}
+
+	/**
+	 * Setter for attributes. We prevent modifying the list of attributes
+	 * because we want to maintain the private field attributeLookup.
+	 * @param inputAttributes
+	 */
+	public void setAttributes(List<FrankAttribute> inputAttributes) {
+		Collections.sort(inputAttributes);
+		this.attributes = Collections.unmodifiableList(inputAttributes);
+		attributeLookup = new HashMap<>();
+		for(FrankAttribute a: attributes) {
+			if(attributeLookup.containsKey(a.getName())) {
+				log.warn(String.format("Frank element [%s] has multiple attributes with name [%s]",
+						fullName, a.getName()));
+			} else {
+				attributeLookup.put(a.getName(), a);
+			}
+		}
+	}
+
+	public List<FrankAttribute> getAttributes(Predicate<? super FrankAttribute> filter) {
+		return attributes.stream().filter(filter).collect(Collectors.toList());
+	}
+
+	/**
+	 * Setter for config children. We prevent modifying the list of config children
+	 * because we want to maintain the private field configChildLookup.
+	 * @param children
+	 */
+	public void setConfigChildren(List<ConfigChild> children) {
+		Collections.sort(children);
+		this.configChildren = Collections.unmodifiableList(children);
+		configChildLookup = new HashMap<>();
+		for(ConfigChild c: children) {
+			ConfigChildKey key = new ConfigChildKey(c);
+			if(configChildLookup.containsKey(key)) {
+				log.warn(String.format("Different config children of Frank element [%s] have the same key", fullName));
+			} else {
+				configChildLookup.put(key, c);
+			}
+		}
+	}
+
+	public List<ConfigChild> getConfigChildren(Predicate<? super ConfigChild> filter) {
+		return configChildren.stream().filter(filter).collect(Collectors.toList());
+	}
+
+	ElementChild findElementChildMatch(ElementChild elementChild) {
+		if(elementChild instanceof FrankAttribute) {
+			return findAttributeMatch((FrankAttribute) elementChild);
+		} else if(elementChild instanceof ConfigChild) {
+			return findConfigChildMatch((ConfigChild) elementChild);
+		} else {
+			throw new IllegalArgumentException(String.format(
+					"Expected a FrankAttribute or ConfigChild, but got a [%s]",
+					elementChild.getClass().getName()));
+		}
+	}
+
+	FrankAttribute findAttributeMatch(FrankAttribute attribute) {
+		return attributeLookup.get(attribute.getName());
+	}
+
+	ConfigChild findConfigChildMatch(ConfigChild configChild) {
+		return configChildLookup.get(new ConfigChildKey(configChild));
+	}
+
+	public <T extends ElementChild> FrankElement getNextAncestor(Predicate<ElementChild> childFilter, Class<T> kind) {
```

Comment:
Dit is eigenlijk: getNextAncestorThatHasChildren(). Dan is het beter om het ook zo te noemen.

Reply:
Gedaan.